### DISCUSSION TITLE
Plot order in Splitting CoveragePlot() with two conditions

### DISCUSSION THREAD

**Discussion by ld2336** [stuart-lab/signac#1136] [Q&A] [answered]
Hi,
  I'd like to generate a coverage plot that shows the chromatin accessibility of several clusters, split between two conditions: Ctrl and KO.
  I found a detailed and useful guide in this discussion:  https://github.com/timoast/signac/discussions/796, even though there's no split.by function in CoveragePlot(). But it will arrange the condition and cluster name in alphabetical order. Like if I need to show some regions in these clusters by this order: Ctrl_Stem, KO_Stem, Ctrl_Neuron, KO_Neuron( one to one comparison),  it actually returns in this way:  Ctrl_Neuron, Ctrl_Stem, KO_Neuron, KO_Stem. Do you know any solution to it?

Thank you so much.

**Comment by timoast** [comment, marked as answer]
You will need to change the factor levels for the cell identities to be the order you want them to appear in the plot. You can do this using the `levels()` function, for example:

```r
library(Signac)
levels(atac_small)
```

```
"1" "0"
```
```r
fpath <- system.file("extdata", "fragments.tsv.gz", package="Signac")
fragments <- CreateFragmentObject(
  path = fpath,
  cells = colnames(atac_small),
  validate.fragments = FALSE
)
Fragments(atac_small) <- fragments

CoveragePlot(object = atac_small, region = c("chr1-713500-714500"))
```

<img width="660" alt="Screen Shot 2022-06-07 at 1 49 18 PM" src="https://user-images.githubusercontent.com/4591688/172305338-ab36fd55-bcf1-4d1d-bbe1-8c683c747ff0.png">

```r
levels(atac_small) <- c("0", "1")
CoveragePlot(object = atac_small, region = c("chr1-713500-714500"))
```
<img width="660" alt="Screen Shot 2022-06-07 at 1 50 16 PM" src="https://user-images.githubusercontent.com/4591688/172305441-b604ac21-19c1-48e0-b942-7022650b6c25.png">

**Reply by ld2336** [reply]
Hi @timoast , I tried your way but the levels() shows the combined cluster name, not in two conditions.
Back to https://github.com/timoast/signac/discussions/796, you solved his problem by creat a new metadata that combines condition and cluster name. 
![133497138-e15f0f83-7b11-42f2-b441-98fe691d2a74](https://user-images.githubusercontent.com/80787141/172422653-99a7190c-bae1-4cf0-b180-65e1b1774662.png)
So how to change the order in this situation, ie. put Saline_1hr_10 on top, PTZ_1hr_10 below?

**Reply by timoast** [reply]
1. Make the new metadata that combines condition and cluster information
2. Set the new metadata as the active identifies using the `Idents()` function
3. Change the factor levels as shown above

**Reply by ld2336** [reply]
Thank you so much for your explanation!
